test/cases/compile_errors/union_init_with_none_or_multiple_fields.zig

```diff
@@ -0,0 +1,37 @@
+const U1 = union {
+    a: u8,
+    b: i8,
+};
+const U2 = union(enum) {
+    a: u8,
+    b: i8,
+};
+fn u1z() void {
+    const x: U1 = .{};
+    _ = x;
+}
+fn u1m() void {
+    const x: U1 = .{ .a = 1, .b = 1 };
+    _ = x;
+}
+fn u2z() void {
+    const x: U2 = .{};
+    _ = x;
+}
+fn u2m() void {
+    const x: U2 = .{ .a = 1, .b = 1 };
+    _ = x;
+}
+export fn entry() void {u1z();u1m();u2z();u2m();}
+// error
+// backend=stage2
+// target=native
+//
+// :9:1: error: cannot initialize multiple union fields at once, unions can only have one active field
```

Comment:
This and `:17:1:` are pointing to the wrong line with an invalid error message and I'd like to get that fixed before merging this. Feel free to ask for help if needed or alternatively I can do it for you.